### PR TITLE
Feature enumerable methods

### PR DESCRIPTION
**Pull request by davidauza-engineer**
Please let me know your comments about my work. Thanks in advance :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,491 @@
...
+        enumerable[key] = value if yield(key, value)
+      end
+    end
+    enumerable
+  end
+
+  def my_all?
+    # Return true if no block is given
+    return true unless block_given?
+
```
Comment (samgaco) [inline README.md]: This method should pass:

```
%w[ant bear cat].all? { |word| word.length >= 3 } #=> true
%w[ant bear cat].all? { |word| word.length >= 4 } #=> false
%w[ant bear cat].all?(/t/)                        #=> false
[1, 2i, 3.14].all?(Numeric)                       #=> true
[nil, true, 99].all?                              #=> false
[].all?                                           #=> true
```

File: README.md [hunk 2]
```diff
@@ -1 +1,491 @@
...
+        boolean = false unless yield(key, value)
+      end
+    end
+    boolean
+  end
+
+  def my_any?
+    # Return true if no block is given
+    return true unless block_given?
+
```
Comment (samgaco) [inline README.md]:
```
%w[ant bear cat].any? { |word| word.length >= 3 } #=> true
%w[ant bear cat].any? { |word| word.length >= 4 } #=> true
%w[ant bear cat].any?(/d/)                        #=> false
[nil, true, 99].any?(Integer)                     #=> true
[nil, true, 99].any?                              #=> true
[].any?                                           #=> false
```

File: README.md [hunk 3]
```diff
@@ -1 +1,491 @@
...
+    end
+    boolean
+  end
+
+  def my_none?
+    # Return false in case no block is provided
+    return false unless block_given?
+
+    boolean = true
+    if self.class == Array
```
Comment (samgaco) [inline README.md]:
```
%w{ant bear cat}.none? { |word| word.length == 5 } #=> true
%w{ant bear cat}.none? { |word| word.length >= 4 } #=> false
%w{ant bear cat}.none?(/d/)                        #=> true
[1, 3.14, 42].none?(Float)                         #=> false
[].none?                                           #=> true
[nil].none?                                        #=> true
[nil, false].none?                                 #=> true
[nil, false, true].none?                           #=> false
```

File: README.md [hunk 4]
```diff
@@ -1 +1,491 @@
...
+        array << yield(key, value)
+      end
+    end
+    array
+  end
+
+  def my_inject(symbol = nil, initial_value = nil)
+    if symbol.class != Symbol
+      temp = symbol
+      symbol = initial_value
```
Comment (samgaco) [inline README.md]:
```
# Same using a block and inject
(5..10).inject { |sum, n| sum + n }            #=> 45
# Same using a block
(5..10).inject(1) { |product, n| product * n } #=> 151200
# find the longest word
longest = %w{ cat sheep bear }.inject do |memo, word|
   memo.length > word.length ? memo : word
end
longest                                        #=> "sheep"
```

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @davidauza-engineer 

- [x] Your methods, my_any?, my_all?, my_none? and my_inject? are still not well implemented, please make sure all of them pass the cases indicated in the comments below.

Remember that  my_any?, my_all?, my_none? , should be able to pass a class, a regulat expression and a pattern as arguments (by pattern we understand anything like "abc" or just a number 1, etc.). And they also should work when no arguments or blocks are given.


- [x] [**SUGGESTION**] Is there a reason for the readme file to have all the code in there? maybe it could be replaced with some explanation that summarizes what is going to be found inside the project.

Please :pray: submit back once that is addressed.

-----------------------------------
* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

File: README.md (hunk 1 above)
Comment (davidauza-engineer) [reply README.md]: All these test cases are correct now 👍

File: README.md (hunk 3 above)
Comment (davidauza-engineer) [reply README.md]: Updated the method so it passes all those test cases 👍

File: README.md (hunk 2 above)
Comment (davidauza-engineer) [reply README.md]: Updated this method to accurately pass all test cases 👍

File: README.md (hunk 4 above)
Comment (davidauza-engineer) [reply README.md]: The method was updated to successfully pass all test cases 👍

**Comment by davidauza-engineer** [comment]
All the issues were successfully fixed, please let me know your comments about the project :)

File: enumerable_methods.rb [hunk 5]
```diff
@@ -0,0 +1,536 @@
...
+        break unless boolean
+      end
+    else
+      my_each do |key, value|
+        boolean = false if yield(key, value)
+        break unless boolean
+      end
+    end
+    boolean
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):030:0> words
=> ["dog", "door", "rod", "blade"]
irb(main):031:0> words.none?(5)
=> true
irb(main):032:0> words.my_none?(5)
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):32
        4: from enumerable_methods.rb:113:in `my_none?'
        3: from enumerable_methods.rb:25:in `my_each'
        2: from enumerable_methods.rb:118:in `block in my_none?'
        1: from enumerable_methods.rb:118:in `<='
TypeError (compared with non class/module)
irb(main):033:0>
```

File: enumerable_methods.rb [hunk 6]
```diff
@@ -0,0 +1,536 @@
...
+        end
+      end
+    else
+      my_each do |key, value|
+        boolean = true if yield(key, value)
+      end
+    end
+
+    boolean
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):025:0> words = %w[dog door rod blade]
=> ["dog", "door", "rod", "blade"]
irb(main):026:0> words.any?('cat')
=> false
irb(main):027:0> words.my_any?('cat')
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):27
        4: from enumerable_methods.rb:89:in `my_any?'
        3: from enumerable_methods.rb:25:in `my_each'
        2: from enumerable_methods.rb:94:in `block in my_any?'
        1: from enumerable_methods.rb:94:in `<='
TypeError (compared with non class/module)
irb(main):028:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):020:0> false_array = [nil, false, nil, false]
=> [nil, false, nil, false]
irb(main):021:0> false_array.any?
=> false
irb(main):022:0> false_array.my_any?
=> true
irb(main):023:0>
```

File: enumerable_methods.rb [hunk 7]
```diff
@@ -0,0 +1,536 @@
...
+          boolean = false unless n.class <= parameter
+        end
+      end
+    else
+      my_each do |key, value|
+        boolean = false unless yield(key, value)
+      end
+    end
+    boolean
+  end
```
Comment (maelfosso) [inline enumerable_methods.rb]:
```ruby
irb(main):015:0> array
=> [8, 4, 5, 0, 0, 1, 2, 3, 8, 3, 4, 6, 2, 1, 0, 2, 8, 8, 5, 0, 4, 8, 4, 8, 1, 2, 6, 0, 0, 6, 8, 7, 1, 0, 5, 7, 5, 8, 5, 6, 3, 4, 1, 3, 5, 4, 5, 8, 8, 5, 3, 1, 2, 7, 0, 2, 0, 0, 3, 1, 0, 8, 3, 7, 2, 3, 6, 1, 0, 4, 0, 3, 7, 6, 4, 6, 3, 1, 0, 5, 4, 4, 7, 3, 3, 2, 2, 4, 1, 2, 5, 4, 0, 3, 3, 3, 0, 1, 4, 2]
irb(main):016:0> array.all?(3)
=> false
irb(main):017:0> array.my_all?(3)
Traceback (most recent call last):
        6: from /home/elvisfosso/.rbenv/versions/2.5.1/bin/irb:11:in `<main>'
        5: from (irb):17
        4: from enumerable_methods.rb:66:in `my_all?'
        3: from enumerable_methods.rb:25:in `my_each'
        2: from enumerable_methods.rb:72:in `block in my_all?'
        1: from enumerable_methods.rb:72:in `<='
TypeError (compared with non class/module)
irb(main):018:0> 
```

Comment (maelfosso) [reply enumerable_methods.rb]:
```ruby
irb(main):010:0> true_array = [1, true, 'hi', []]
=> [1, true, "hi", []]
irb(main):011:0> true_array.all?
=> true
irb(main):012:0> true_array.my_all?
=> false
irb(main):013:0>
```

**Review by maelfosso — CHANGES_REQUESTED** [review]
# Changes required

Hello @davidauza-engineer 

Really awesome !!! I really clap for the improvements :clap: 
You're about to finish, only 5 bugs to fix, please. Read the comment left below to know more about the errors.

Kindly make the changes and submit a code review request.
Thanks

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

File: enumerable_methods.rb (hunk 5 above)
Comment (davidauza-engineer) [reply enumerable_methods.rb]: This issue was successfully fixed 👍

File: enumerable_methods.rb (hunk 6 above)
Comment (davidauza-engineer) [reply enumerable_methods.rb]: Both issues were successfully fixed 👍

File: enumerable_methods.rb (hunk 7 above)
Comment (davidauza-engineer) [reply enumerable_methods.rb]: Both issues were successfully fixed 👍

**Comment by davidauza-engineer** [comment]
All the issues were successfully fixed. Please let me know your comments about the project :)

**Review by Redvanisation — APPROVED** [review]
# Project approved ✔️ 

### Hello @davidauza-engineer  🌟 

### _Nice work on this project, it seems that you have met the requirements so I am approving your project. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀

**Comment by davidauza-engineer** [comment]
Thank you so much guys @samgaco , @maelfosso , @Redvanisation , all the reviews were so helpful 👍

**Pull request merged**